nixos/modules/services/networking/tinydns.nix

```diff
@@ -37,6 +37,7 @@ with lib;
     systemd.services.tinydns = {
       description = "djbdns tinydns server";
       wantedBy = [ "multi-user.target" ];
+      after = [ "network.target" ];
```

Comment:
You should never use `network.target`: it's meant to be pulled in by services providing the network link/setup, like dhcpcd. Use `network-online.target` instead. See https://github.com/NixOS/nixpkgs/issues/50930#issuecomment-441775955

Reply:
Does tinydns fail horribly during startup if there's no default route? Otherwise, systemd docs explicitly mention we should go after `network.target`, not `network-online.target`:

> Units that strictly require a configured network connection should pull in network-online.target (via a Wants= type dependency) and order themselves after it. This target unit is intended to pull in a service that delays further execution until the network is sufficiently set up. What precisely this requires is left to the implementation of the network managing service.

Reply:
> Does tinydns fail horribly during startup if there's no default route?

Probably not but it's logically not part of setting up the network link and needs an internet connection.

Reply:
We don't set `wants`, or make it `partOf` `network.target` - we only set an `after`.

Setting `After=network-online.target` would mean we wait with starting tinydns until there's a default route. This means if you're enabling tinydns on your laptop, you don't reach `multi-user.target` until you get an internet connection.

I don't think that's necessary, but tinydns needs to wait until there's a network interface available in the system, so setting `After=network.target` should be right. :confused:

Reply:
Oh, you are right: I was confusing `After` with `Wants`, sorry.

Reply:
No worries!